analyzer/src/traversal/assignments.rs

```diff
@@ -13,16 +13,21 @@ use std::rc::Rc;
 /// e.g. `foo[42] = "bar"`, `self.foo[42] = "bar"`, `foo = 42`
 pub fn assign(
     scope: Shared<BlockScope>,
-    context: Shared<Context>,
+    context: &mut Context,
```

Comment:
Interesting! I remember seeing the discussion on discord. I want to read up and educate myself about the details of why this is possible here :)

Reply:
There aren't any tricky ownership situations involving Context, so it was just a straight search-and-replace. Scope is another story, and will probably stay Rc for now.